tests/api/test_statements_post.py

```diff
@@ -218,3 +222,100 @@ def test_post_statements_list_with_duplicate_of_existing_statement(
     es.indices.refresh()
     hits = es.search(index=ES_TEST_INDEX)["hits"]["hits"]
     assert len(hits) == 1
+
+
+@pytest.mark.asyncio
+@mock.patch("ralph.api.routers.statements.ES_CLIENT", ESDatabase(index=ES_TEST_INDEX))
+async def test_post_statements_list_with_statement_forwarding(
+    monkeypatch, auth_credentials, es, lrs  # pylint: disable=unused-argument
+):
+    """Tests the xAPI forwarding functionality given two ralph instances - a forwarding
+    instance and a receiving instance. When the forwarding instance receives a valid
+    xAPI statement it should store and forward it to the receiving instance which in
+    turn should store it too.
+    """
+
+    # Set-up forwarding and receiving Elasticsearch clients and indexes.
+    es_index_forwarding = ES_TEST_INDEX
+    es_client_forwarding = ESDatabase(index=es_index_forwarding)
+    es_index_receiving = f"{ES_TEST_INDEX}_2"
+    es_client_receiving = ESDatabase(index=es_index_receiving)
+    try:
+        es_client_receiving.client.indices.create(index=es_index_receiving)
+    except BadRequestError:
+        # Index already exists.
+        pass
+
+    # Set-up receiving LRS client.
+    with monkeypatch.context() as receiving_patch:
+        # Receiving client should not forward statements.
+        receiving_patch.setattr("ralph.api.forwarding.VALID_XAPI_FORWARDING", [])
+        # Receiving client should use the receiving Elasticsearch client for storage.
+        receiving_patch.setattr(
+            "ralph.api.routers.statements.ES_CLIENT", es_client_receiving
+        )
+        lrs_context = lrs(app)
+        # Start receiving LRS client.
+        await lrs_context.__aenter__()  # pylint: disable=unnecessary-dunder-call
+
+    # Set-up forwarding client.
+    with monkeypatch.context() as forwarding_patch:
+        # Forwarding client should forward statements to receiving client.
+        forwarding_patch.setattr(
+            "ralph.api.forwarding.VALID_XAPI_FORWARDING",
+            [
+                ForwardingConfiguration(
+                    url="http://localhost:8101/xAPI/statements/",
+                    is_active=True,
+                    basic_username="ralph",
+                    basic_password="admin",
+                    max_retries=1,
+                    timeout=10,
```

Comment:
Could it be (an) input configuration parameter(s) ?

Reply:
Good point) Indeed we could define this part using hypothesis; however, I'm hesitating as now the test has changed and already contains nine arguments.